slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -72,13 +74,55 @@ void MapBuilder::LoadMapFromFile(std::string map_filename,
                 << trajectory_ids_pair.second;
 }
 
-void MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
+bool MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
                                const std::string filename_with_timestamp) {
     bool ok = map_builder_->SerializeStateToFile(include_unfinished_submaps,
                                                  filename_with_timestamp);
     if (!ok) {
         LOG(ERROR) << "Saving the map to pbstream failed.";
     }
+    return ok;
+}
+
+std::string MapBuilder::SaveMapToStream(std::string filename,
```

Comment:
[opt] Returning a string error is unusual for C++. Usually you would throw an exception and catch it at the level above. But I think this is fine.